### ISSUE TITLE
When multiple service nodes are defined, prescript are executed on each service node

### ISSUE THREAD

**Issue by kcgthb** [xcat2/xcat-core#6236]
Hi,

When multiple service nodes are defined for a node in the `noderes` table, `prescripts` seem to be executed once on each service node.
The `noderes` documentation says:
```
servicenode
    A comma separated list of node names (as known by the management node) that 
    provides most services for this node. The first service node on the list that is
    accessible will be used.  The 2nd node on the list is generally considered to be the
    backup service node for this node when running commands like snmove.
```
The `prescript`  behavior doesn't seem to match `The first service node on the list that is accessible will be used.`

Is this expected?

Thanks!

**Comment by robin2008** [comment]
@kcgthb 
Yes, this is current expected behavior, could we know any of your requirements?

Here the `first` and `second` only for kinds of commands, like rpower/rcons.  When the first is done, then xcat will try to dispatch it to second.

While for `prescripts`, this is executed during `nodeset` to customize the pxe/kickstart or something else on local host. So it should be executed on all service nodes which need to handle `nodeset`. 
And which service nodes are requires to handle `nodeset` is  controlled by `disjointdhcps`
```
                  disjointdhcps:  If set to '1', the .leases file on a service node only contains
                                  the nodes it manages. And when 'sharedtftp' is disabled, nodeset handles
                                  boot loader configuration on a service node only for the nodes it manages.
                                  The default value is '0'. It means include all the nodes in the subnet.
```

**Comment by kcgthb** [comment]
@robin2008 
Since the documentation says `The first service node on the list that is accessible will be used`, that kind of implies that once a accessible service node has been found in the list, the others won't be used.

So if this is indeed the expected behavior, the documentation should likely be modified to something like this: `All the available service nodes on the list will be used.`

I'm actually not sure to see in what case it would be helpful to run scripts through a 2nd service node if they already ran on the 1st one, especially in the case of `disjointdhcps = 0`: because in that case, multiple service nodes would more likely be used for redundancy, since they all address all the nodes.

**Comment by robin2008** [comment]
Well,  the document need to be refine and hightlight the difference for using this attribute between two kinds of commands:
- management view:  `remote control(rpower), updatenode(xdcp, xdsh)`
- provisioning view: rinstall/makedhcp/nodeset

For the 1st case, just as the document said,  `The first service node on the list that is accessible will be used`
while for the 2nd case, it should be `All the available service nodes on the list will be used.`

**Comment by kcgthb** [comment]
I guess there's a 3rd case, which is using `prescripts`, for which you don't necessarily want to run them multiple times, once per service node. So maybe a flag could be introduced for this?

I can see how you want all your service nodes to be updated wrt. DHCP, PXE when they can all potentially server the same node, but one can expect to run prescripts only once per node being deployed, so that multiple execution could be an issue in that case.
